source/slang/parser.cpp

```diff
@@ -1835,24 +1835,76 @@ namespace Slang
         }
     }
 
-    //
-    // layout-semantic ::= (register | packoffset) '(' register-name component-mask? ')'
-    // register-name ::= identifier
-    // component-mask ::= '.' identifier
-    //
-    static void ParseHLSLLayoutSemantic(
-        Parser*				parser,
-        HLSLLayoutSemantic*	semantic)
+    /// Parse the "register name" part of a `register` or `packoffset` semantic.
+    ///
+    /// The syntax matched is:
+    ///
+    ///     register-name-and-component-mask ::= register-name component-mask?
+    ///     register-name ::= identifier
+    ///     component-mask ::= '.' identifier
+    ///
+    static void parseHLSLRegisterNameAndOptionalComponentMask(
+        Parser*             parser,
+        HLSLLayoutSemantic* semantic)
     {
-        semantic->name = parser->ReadToken(TokenType::Identifier);
-
-        parser->ReadToken(TokenType::LParent);
         semantic->registerName = parser->ReadToken(TokenType::Identifier);
         if (AdvanceIf(parser, TokenType::Dot))
         {
             semantic->componentMask = parser->ReadToken(TokenType::Identifier);
         }
+    }
+
+    /// Parse an HLSL `register` semantic.
+    ///
+    /// The syntax matched is:
+    ///
+    ///     register-semantic ::= 'register' '(' register-name-and-component-mask register-space? ')'
+    ///     register-space ::= ',' identifier
+    ///
+    static void parseHLSLRegisterSemantic(
+        Parser*                 parser,
+        HLSLRegisterSemantic*   semantic)
+    {
+        // Read the `register` keyword
+        semantic->name = parser->ReadToken(TokenType::Identifier);
+
+        // Expect a parenthized list of additional arguments
+        parser->ReadToken(TokenType::LParent);
```

Comment:
LParent  - had to double take ;) LBracket? LParenthesis?

Reply:
We probably could use a cleanup pass on some of our token naming. I'm used to "paren" for `()`, "curly" or "brace" for `{}`, and "square" or "bracket" for `[]`, and I'm not a big fan of "parent" here.

I'd also advocate for a switch from `TokenType` to `TokenCode`, but that's an even more minor point.